FFmpegInterop/Source/MediaSampleProvider.cpp

```diff
@@ -223,17 +223,17 @@ HRESULT FFmpegInterop::MediaSampleProvider::GetNextPacket(DataWriter ^ writer, L
 			DebugMessage(L"Saving m_startOffset\n");
 
 			//in some real-time streams framePts is less than 0 so we need to make sure m_startOffset is never negative
-			m_startOffset = framePts < 0 ? 0 : framePts;
+			m_startOffset = m_pAvFormatCtx->streams[m_streamIndex]->start_time < 0 ? 0 : m_pAvFormatCtx->streams[m_streamIndex]->start_time;
```

Comment:
How about using (positive) framePts as fallback if we do not have start_time? Or can we be sure that start_time is there?

Reply:
Hi,

not sure if it can help but https://github.com/Microsoft/FFmpegInterop/commit/fd353fe03bfe40cd3fdae8df86b644d5b6be083f was tested a lot and proved to be working. 
I don't know what made @khouzam choose another implementation and I won't link to the original thread because it became a real mess because of other non related issues !

Reply:
I had another way of fixing this. But this PR works. the only thing I would change is to move the resolving of start time in the AllocateResources() method

Reply:
I don't know why only positive offsets were taken into account in the first place. Doesn't it make sense to apply the start offset whatever the value different of AV_NOPTS_VALUE is ? 
I never encountered the case myself but I think start_time would be AV_NOPTS_VALUE if it could not be set by ffmpeg.

Reply:
FYI, I encountered negative start_time value in a MPEG-TS stream dump that contains a PTS rollover.